sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -261,6 +269,10 @@ createTableHeader
     : CREATE TEMPORARY? EXTERNAL? TABLE (IF NOT EXISTS)? multipartIdentifier
     ;
 
+replaceTableHeader
```

Comment:
I'm worried about creating new SQL syntax in Spark. AFAIK a similar syntax is `CREATE OR REPLACE TABLE`, which is implemented in [DB2](https://www.ibm.com/support/knowledgecenter/en/ssw_ibm_i_72/sqlp/rbafyreplacetable.htm) and [google BigQuery](https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language).

This is not a standard SQL syntax, so it's not surprising to see that Oracle doesn't support it. If Spark want a API for replace table, I think it's more reasonable to follow DB2 and BigQuery here.

Reply:
I'm fine using `[CREATE OR] REPLACE TABLE` instead of `REPLACE TABLE [IF NOT EXISTS]`. I think that is better syntax.

Reply:
So do we remove the existing support for the `IF NOT EXISTS` syntax in the SQL syntax? I think we'd want to avoid breaking existing SQL queries even in the 3.0 release.

Or do we support:

```
CREATE (OR REPLACE)? TABLE ... (IF NOT EXISTS)?
```

and then throw `AnalysisException` if: `CREATE OR REPLACE TABLE ... IF NOT EXISTS`?

Reply:
> I'm fine using `[CREATE OR] REPLACE TABLE` instead of `REPLACE TABLE [IF NOT EXISTS]`. I think that is better syntax.

We don't support `REPLACE... IF NOT EXISTS` in this PR. It's only either `CREATE TABLE IF NOT EXISTS` or `REPLACE TABLE`. I don't see any reason to include `CREATE OR REPLACE` table if we don't want to replace the `IF NOT EXISTS` parameter in existing spark-sql, but removing support for `IF NOT EXISTS` risks breaking existing SQL workflows as I've described above.

Reply:
I'd say let's remove `IF NOT EXISTS` from the parser, but just for `REPLACE TABLE`.

Then, we should add an option to add `CREATE OR`, to the start, which would set an `orCreate` flag to true. If that is true, and the table doesn't exist, then create the table instead of replacing it. If that is false, then throw an exception that the table doesn't exist.

Reply:
I think that makes sense, though the current implementation doesn't have a rule supporting `REPLACE TABLE IF NOT EXISTS`.

Reply:
It it isn't supported, then we can throw an exception.

Reply:
Since that clause isn't supported in the `SqlBase.g4` file itself, I'd expect the parser to throw an exception for us.